lumicks/pylake/tests/test_fdcurve.py

```diff
@@ -175,3 +178,23 @@ def test_copy_behaviour_with_offset(h5_file):
     is because with_offset may not deepcopy the handle it holds to the parent file."""
     fd1 = make_mock_fd(force=[1, 2, 3], distance=[2.0, 0.0, 4.0], start=0, file=h5_file)
     fd1.with_offset(distance_offset=-1.0)
+
+
+def test_with_channels(fd_h5_file):
+    f = File.from_h5py(fd_h5_file)
+    fd = f.fdcurves["fd1"]
+    assert fd._primary_force_channel == "2"
+    assert fd._primary_distance_channel == "1"
+
+    fd2 = fd.with_channels(force="1", distance="2")
+    assert fd2._primary_force_channel == "1"
+    assert fd2._primary_distance_channel == "2"
+
+
+def test_with_baseline_correction(fd_h5_file):
+    f = File.from_h5py(fd_h5_file)
+    fd = f.fdcurves["fd1"]
+
+    fd2 = fd.with_baseline_correction_x()
+    assert not np.all(np.equal(fd.f.data, fd2.f.data))
```

Comment:
This is great.  Since we have the coefficients I'd also add at least one force entry's manual calculation that reverses the corrected force back to the original, then compare that with the `fd.fd.data`, something like:
```
# get the polynomial
p = np.poly1d(f.fdcurves.metadata.coefficients) # Not sure about the actual coefficients variable name

# calculate the original force using the polynomial
x0 = fd.d.data[0]
reverted_f_data_entry = x0 - p(x0)

# compare the manually calculated force against the original
assert np.equal(reverted_f_data_entry, fd.f.data[0])
```

Reply:
the coefficients are being pushed off to another ticket to try to keep these PRs from becoming too massive. Right now we just read the corrected data channel directly, there's no import for the metadata. I agree this test should be included when that feature is implemented though